crates/core/src/secrets.rs

```diff
@@ -0,0 +1,22 @@
+use std::fmt::{Debug, Formatter, Result};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Clone)]
+// This tagging allows deserializers to know whether the secret is a string or bytes.
+// This is especially necessary for languages where strings and bytes are treated very similarly.
+#[serde(tag = "kind", content = "value")]
+pub enum SecretValue {
+    String(String),
```

Comment:
I did see further down that this type implements some secrecy stuff so maybe this doesn't matter?

Reply:
There are a few different enums/structs with the name `SecretValue`, so perhaps it would be best to rename some of these for clarity. I don't implement the secrecy crate here since I didn't want the representation to change and be complicated to work with from, say, the Go provider SDK, but maybe if I implement serialize it wouldn't matter. 

If we're cool with `core` depending on the secrecy crate, I can give it a shot. Slightly concerned with the tagged enum piece of it bit I'm sure I could get it working